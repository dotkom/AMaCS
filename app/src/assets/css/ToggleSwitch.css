.container {
  display: inline-flex;
   
  background-color: #ffffff;
  border: 2px solid #636363;
  border-radius: 16px;
  justify-content: center;
  user-select: none;
  cursor: default;
}

.container:focus {
  outline: none;
}

.container:disabled {
  background: #f0f0f0;
  border-color: #b8b8b8;
}

.slider {
  margin: 0.07rem -0.17rem;
  margin-right: 1.5rem;
  align-self: left;
  width: 1.5rem;
  height: 1.5rem;
  transition: .4s;
  border-radius: 100%;
  background: #D06969;
}

.container:hover .slider {
  background: #a55555;
}

.checked {
  margin-left: 1.5rem;
  margin-right: -0.17rem;
  background: #0060A3;
}

.container:hover .checked {
  background: #004b80;
}

.container:disabled .slider {
  background: #ababab;
}

.icon {
  stroke: white;
  stroke-width: 3;
}
