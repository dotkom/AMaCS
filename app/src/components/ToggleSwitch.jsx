import React from "react";
import classNames from 'classnames';

import Cross from 'assets/images/cross.svg';
import Check from 'assets/images/check.svg';
import _s from 'assets/css/ToggleSwitch.css';

function ToggleSwitch({ checked, disabled, onChange }) {
  const SvgIcon = checked ? Check : Cross;
  return(
    <button
      disabled={disabled}
      onClick={onChange}
      className={_s.container}
    >
      <div className={classNames(_s.slider, { [_s.checked]: checked })}>
        <span className={_s.icon}>
          <SvgIcon width={12} />
        </span>
      </div>
    </button>
  )
}


ToggleSwitch.defaultProps = {
  checked: false
}

export default ToggleSwitch;
