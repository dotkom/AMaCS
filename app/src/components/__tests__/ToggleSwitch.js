import React from 'react';
import { shallow, mount } from 'enzyme';

import ToggleSwitch from '../ToggleSwitch';

describe('ToggleSwitch.jsx', () => {
  
  const onChangeMock = jest.fn();


  afterEach(() => {
    jest.clearAllMocks();
  });


  it('renders correctly', () => {
    const wrapper = shallow(
      <ToggleSwitch />
    );
    expect(wrapper).toMatchSnapshot();
  });

  it('renders checked', () => {
    const wrapper = shallow(
      <ToggleSwitch checked />
    );
    expect(wrapper).toMatchSnapshot();
  });

  it('toggles checked when clicked', () => {
    const wrapper = mount(
      <ToggleSwitch />
    );
    expect(wrapper.state('checked')).toBe(false);
    wrapper.find('button').simulate('click');
    expect(wrapper.state('checked')).toBe(true);
    expect(wrapper).toMatchSnapshot();
  });


  it('toggles checked when clicked and component updates as an uncontrolled component', () => {
    const wrapper = mount(
      <ToggleSwitch />
    );
    expect(wrapper.state('checked')).toBe(false);
    wrapper.find('button').simulate('click');
    expect(wrapper.state('checked')).toBe(true);
    wrapper.update();
    expect(wrapper.state('checked')).toBe(true);
    expect(wrapper).toMatchSnapshot();
  });

  it('toggles un-checked when clicked and component updates as a controlled component', () => {
    const checked=true;
    const wrapper = mount(
      <ToggleSwitch checked={checked} />
    );
    expect(wrapper.state('checked')).toBe(checked);
    wrapper.find('button').simulate('click');
    expect(wrapper.state('checked')).toBe(!checked);
    wrapper.update();
    expect(wrapper.state('checked')).toBe(checked);
    expect(wrapper).toMatchSnapshot();
  });

  it('toggles and checks if onChange is called with correct values', () => {
    const checked=true;
    const wrapper = mount(
      <ToggleSwitch 
        checked={checked} 
        onChange={onChangeMock}
      />
    );

    wrapper.find('button').simulate('click');
    expect(onChangeMock).toHaveBeenCalledWith(!checked);
    wrapper.find('button').simulate('click');
    expect(onChangeMock).toHaveBeenCalledWith(checked);
    expect(wrapper).toMatchSnapshot();
  });
});
