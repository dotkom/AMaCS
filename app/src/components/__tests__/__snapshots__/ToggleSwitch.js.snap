// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ToggleSwitch.jsx renders checked 1`] = `
<button
  className="container"
>
  <div
    className="slider checked"
  >
    <span
      className="icon"
    >
      <SVGMockComponent
        width={12}
      />
    </span>
  </div>
</button>
`;

exports[`ToggleSwitch.jsx renders correctly 1`] = `
<button
  className="container"
>
  <div
    className="slider"
  >
    <span
      className="icon"
    >
      <SVGMockComponent
        width={12}
      />
    </span>
  </div>
</button>
`;
